purchase_packaging/models/purchase_order_line.py

```diff
@@ -0,0 +1,179 @@
+        if self.product_id:
+            supplier = self._get_product_seller()
+            if supplier:
+                product_purchase_uom = supplier.min_qty_uom_id
+                uom_obj = self.env['product.uom']
+                from_uom = uom_obj.search(
+                    [('category_id', '=',
+                      product_purchase_uom.category_id.id),
+                     ('uom_type', '=', 'reference')], limit=1)
+                self.product_purchase_qty = from_uom._compute_quantity(
+                    self.product_qty,
+                    product_purchase_uom)
+                self.product_purchase_uom_id = product_purchase_uom.id
+        else:
+            self.product_purchase_qty = self.product_qty
+
+    @api.onchange("packaging_id")
+    def _onchange_packaging_id(self):
+        if self.packaging_id:
+            self.product_uom = self.packaging_id.uom_id
+
+    @api.onchange('product_id')
+    def onchange_product_id(self):
+        """ set domain on product_purchase_uom_id and packaging_id
+            set the first packagigng, purchase_uom and purchase_qty
+        """
+        domain = {}
+        # call default implementation
+        # restore default values
+        defaults = self.default_get(
+            ['packaging_id', 'product_purchase_uom_id'])
+        self.packaging_id = self.packaging_id.browse(
+            defaults.get('packaging_id', []))
+        self.product_purchase_uom_id = self.product_purchase_uom_id.browse(
+            defaults.get('product_purchase_uom_id', []))
+        # add default domains
+        if self.product_id and self.partner_id:
+            domain['packaging_id'] = [
+                ('id', 'in', self.product_id.mapped(
+                    'seller_ids.packaging_id.id'))]
+            domain['product_purchase_uom_id'] = \
+                [('id', 'in', self.product_id.mapped(
+                    'seller_ids.min_qty_uom_id.id'))]
+        res = super(PurchaseOrderLine, self).onchange_product_id()
+        if self.product_id:
+            supplier = self._get_product_seller()
+        else:
+            supplier = self.product_id.seller_ids.browse([])
+        if supplier.product_uom:
+            # use the uom from the suppleir
+            self.product_uom = supplier.product_uom
+        if supplier.min_qty_uom_id:
+            # if the supplier requires some min qty/uom,
+            self.product_purchase_qty = supplier.min_qty
+            self.product_purchase_uom_id = supplier.min_qty_uom_id
+            domain['product_purchase_uom_id'] = \
+                [('id', '=', supplier.min_qty_uom_id.id)]
+            to_uom = self.env['product.uom'].search([
+                ('category_id', '=',
+                 supplier.min_qty_uom_id.category_id.id),
+                ('uom_type', '=', 'reference')], limit=1)
+            to_uom = to_uom and to_uom[0]
+            self.product_qty = supplier.min_qty_uom_id._compute_quantity(
+                supplier.min_qty, to_uom
```

Comment:
You need to convert to the uom of the supplier, if one was found in https://github.com/OCA/stock-logistics-warehouse/pull/259/files#diff-0b094619c99f4fabbe358caff73c8626R119. 

For example, if you have a minimum of 1 Half Dozen's of a product with packaging of Dozen's. When you order this product you will get by default 1 Half Dozen as Purchase UoM and 1 Dozen as Uom. You should get 1 Half Dozen and 0.5 Dozen.

Reply:
@mreficent Purchase unit of measure in this case are a coefficent of the product unit of measure.

So, if your minimal purchase quantity is for example 0.5 x Dozen (Purchase UOM) of Dozen (Product UOM), the purchase order line will show in product_quantity 6 x Dozen (which is 0.5 x 12 x 12)

If your minimal quantity is 0.5 x Unit (Purchase UOM) of Dozen (Product UOM), the purhase order line will show in product quantity 0.5 x Dozen (which is 0.5 x 1 x 12).

If your product unit of measure is Unit, you can have a minimal quantity of 0.5 x Dozen (Purchase UOM) of Unit (Product UOM), that will be translated in 6 x Unit.